flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/channel/ChannelStateWriterImplTest.java

```diff
@@ -121,15 +141,28 @@ public void testAbortIgnoresMissing() throws Exception {
         runWithSyncWorker(this::callAbort);
     }
 
+    @Test
+    public void testOldCheckpointIsAborted() {
```

Comment:
Aren't we missing a test coverage for the main point of this bug? That the following sequence:
```
callAbort(1, ...);
callStart(1, ...);
```
also leads to a completed `ChannelStateWriteResult`? Previously that would create an uncompleted `ChannelStateWriteResult`, that was causing a deadlocks in FLINK-26803 right?